panther_manager/src/manager_bt_node.cpp

```diff
@@ -49,6 +111,50 @@ ManagerNode::ManagerNode(std::shared_ptr<ros::NodeHandle> nh) : nh_(std::move(nh
   ROS_INFO("[%s] Node started", node_name_.c_str());
 }
 
+void ManagerNode::setup_behaviortree(
+  BT::Tree & tree, BT::NodeConfig & config, const std::string tree_name,
+  std::map<std::string, std::any> bb_values)
+{
+  config.blackboard = BT::Blackboard::create();
+  // update blackboard
+  config.blackboard->set("nh", nh_);
+  for (auto & item : bb_values) {
+    const std::type_info & type = item.second.type();
+    if (type == typeid(bool)) {
+      config.blackboard->set<bool>(item.first, std::any_cast<bool>(item.second));
+    } else if (type == typeid(int)) {
+      config.blackboard->set<int>(item.first, std::any_cast<int>(item.second));
+    } else if (type == typeid(float)) {
+      config.blackboard->set<float>(item.first, std::any_cast<float>(item.second));
+    } else if (type == typeid(double)) {
+      config.blackboard->set<double>(item.first, std::any_cast<double>(item.second));
+    } else if (type == typeid(long)) {
+      config.blackboard->set<long>(item.first, std::any_cast<long>(item.second));
+    } else if (type == typeid(const char *)) {
+      config.blackboard->set<std::string>(item.first, std::any_cast<const char *>(item.second));
+    } else {
+      throw std::invalid_argument(
+        "Invalid type for blackboard entry. Valid types are: bool, int, float, double, long,"
+        " const char*");
+    }
```

Comment:
Unfortunately it is not possible to replace it with switch, because `typeid()` function is not a constant expression. Also, I played with templates for a bit and haven't found any solution. As bad as this "ifish" code looks a can't see a better solution.